overlord/snapstate/handlers.go

```diff
@@ -1355,6 +1355,28 @@ func (m *SnapManager) restoreUnlinkOnError(t *state.Task, info *snap.Info, tm ti
 	return err
 }
 
+func onRefreshInhibitionTimeout(t *state.Task, snapName string) error {
+	chg := t.Change()
+
+	var data map[string]interface{}
+	err := chg.Get("api-data", &data)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+	if len(data) == 0 {
+		data = make(map[string]interface{})
+	}
+
+	cur, _ := data["refresh-forced"].([]interface{})
+	cur = append(cur, snapName)
+	data["refresh-forced"] = cur
```

Comment:
Meh, a bit unhappy that we still need to do it. I'll see if I can propose something to tweak this

Reply:
is this about the type casting or setting updating "api-data" inside tasks in general?

Reply:
Yes, type assertions. We're basically treating all of it as a loose bag of whatever objects, basically `map[string]any` all the time. The places where the respective values are used already know the exact type and thus could benefit from a helper which marshals/unmarshals with type checking (with the exception where we may want to patch stuff at startup but that's another issue).